src/Components/access-requests-wizard/AccessRequestsWizard.js

```diff
@@ -0,0 +1,284 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
+import {
+  Bullseye,
+  Button,
+  capitalize,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  Modal,
+  ModalVariant,
+  Spinner,
+  Title,
+  Wizard,
+} from '@patternfly/react-core';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
+import FormRenderer from '@data-driven-forms/react-form-renderer/form-renderer';
+import Pf4FormTemplate from '@data-driven-forms/pf4-component-mapper/form-template';
+import componentMapper from '@data-driven-forms/pf4-component-mapper/component-mapper';
+import SelectRoles from './SelectRoles';
+import ReviewDetails from './ReviewDetails';
+import AccessDuration from './AccessDuration';
+import SetName from './SetName';
+import apiInstance from '../../Helpers/apiInstance';
+import schemaBuilder, {
+  ACCESS_FROM,
+  ACCESS_TO,
+  ACCOUNT_NUMBER,
+  ORG_ID,
+  SELECTED_ROLES,
+} from './schema';
+
+const invalidAccountTitle = 'Invalid Account number';
+
+const FormTemplate = (props) => (
+  <Pf4FormTemplate {...props} showFormControls={false} />
+);
+
+export const mapperExtension = {
+  'set-name': SetName,
+  'access-duration': AccessDuration,
+  'select-roles': SelectRoles,
+  'review-details': ReviewDetails,
+};
+
+const AccessRequestsWizard = ({ requestId, variant, onClose }) => {
+  const dispatch = useDispatch();
+  const isEdit = variant === 'edit';
+  const initialValues = useRef();
```

Comment:
Any reason why ref instead of state? I assume you have to avoid re-renders?

Reply:
Good point, state will be better - pushed the update